.github/workflows/review_and_testing.yml

```diff
@@ -1,10 +1,33 @@
-name: testing
+name: Review & Test
+
+permissions:
+  contents: read
+  pull-requests: write
 
 on:
   pull_request:
     branches: [main]
+    
+concurrency:
+  group:
+    ${{ github.repository }}-${{ github.event.number || github.head_ref ||
+    github.sha }}-${{ github.workflow }}-${{ github.event_name ==
+    'pull_request_review_comment' && 'pr_comment' || 'pr' }}
+  cancel-in-progress: ${{ github.event_name != 'pull_request_review_comment' }}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The concurrency settings ensure that only one instance of this workflow runs at a time. This can help prevent unnecessary resource usage if multiple instances of the same workflow are triggered simultaneously. However, please make sure this behavior is desired as it could potentially delay feedback on pull requests if many are opened around the same time.

```diff
  concurrency:
    group:
      ${{ github.repository }}-${{ github.event.number || github.head_ref ||
      github.sha }}-${{ github.workflow }}-${{ github.event_name ==
      'pull_request_review_comment' && 'pr_comment' || 'pr' }}
    cancel-in-progress: ${{ github.event_name != 'pull_request_review_comment' }}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->